sys/include/net/ieee802154/radio.h

```diff
@@ -0,0 +1,1124 @@
+
+    /**
+     * @brief Request Stand-Alone Clear Channel Assessment
+     *
+     * @pre the state is @ref IEEE802154_TRX_STATE_RX_ON
+     *
+     * @note @ref ieee802154_radio_ops::confirm_cca MUST be used to
+     * finish the CCA procedure and get the channel status.
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     *
+     * @post the state is @ref IEEE802154_TRX_STATE_RX_ON
+     *
+     * @return 0 if request was OK
+     * @return -EAGAIN if the request cannot be performed immediately.
+     * @return negative errno on error
+     */
+    int (*request_cca)(ieee802154_dev_t *dev);
+
+    /**
+     * @brief Confirmation function for @ref ieee802154_radio_ops::request_cca
+     *
+     * This function must be called to finish the CCA procedure.  This
+     * function should be called on @ref IEEE802154_RADIO_CONFIRM_CCA,
+     * If no interrupt is available, this function can be polled.
+     *
+     * @pre call to @ref ieee802154_radio_ops::request_cca was successful.
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     *
+     * @return positive number if the channel is clear
+     * @return 0 if the channel is busy
+     * @return -EAGAIN if the CCA procedure hasn't finished.
+     * @return negative errno on error
+     */
+    int (*confirm_cca)(ieee802154_dev_t *dev);
+
+    /**
+     * @brief Get a cap from the radio
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     * @param cap cap to be checked
+     *
+     * @return true if the radio supports the cap
+     * @return false otherwise
+     */
+    bool (*get_cap)(ieee802154_dev_t *dev, ieee802154_rf_caps_t cap);
+
+    /**
+     * @brief Set the threshold for the Energy Detection (first mode of CCA)
+     *
+     * @pre the device is on
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     * @param[in] threshold the threshold in dBm.
+     *
+     * @return 0 on success
+     * @return negative errno on error
+     */
+    int (*set_cca_threshold)(ieee802154_dev_t *dev, int8_t threshold);
+
+    /**
+     * @brief Set CCA mode
+     *
+     * All radios MUST at least implement the first CCA mode (ED Threshold).
+     *
+     * @pre the device is on
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     * @param[in] mode the CCA mode
+     *
+     * @return 0 on success
+     * @return -ENOTSUP if the mode is not supported
+     * @return negative errno on error
+     */
+    int (*set_cca_mode)(ieee802154_dev_t *dev, ieee802154_cca_mode_t mode);
+
+    /**
+     * @brief Set IEEE802.15.4 PHY configuration (channel, TX power)
+     *
+     * This function SHOULD NOT validate the PHY configurations unless
+     * it's specific to the device. The upper layer is responsible of all kind
+     * of validations.
+     * In case a configuration is not valid (e.g parameters out of range), this
+     * function should return -EINVAL
+     *
+     * @pre the device is on
+     * @post the upper layer calls @ref
```

Comment:
Let's remove this post condition. I think it's always possible to ensure the radio will still be working in the corresponding state after calling this function.

Otherwise it becomes hard for the implementation of some network stacks (e.g the OpenThread contrib would need to keep track of the last state when calling this function).

Reply:
Yea the operating state should not change by calling this funcion.